### ISSUE TITLE
Cannot open Diangnostics panel

### ISSUE THREAD

**Issue by stefaneidelloth** [krassowski/jupyterlab-lsp#385]
jupyterlab-lsp shows pep 8 issues but I am not able to find/open "show diagnostics panel" from command palette:
![image](https://user-images.githubusercontent.com/12675339/97104111-8fa66000-16b1-11eb-8eb1-7bd823be66f5.png)

=> Is there another way to show diagnostics panel, e.g. key shortcut?
=> How can I find out more about that issue? Is there some kind of log of jupyterlab-lsp?

Here is the output of jupyter labextension list

```
root@ead842543f65:/home/test# jupyter labextension list
JupyterLab v2.2.8
Known labextensions:
   app dir: /usr/local/share/jupyter/lab
        @jupyterlab/git v0.22.1  enabled  OK
        @krassowski/jupyterlab-lsp v2.0.7  enabled  OK
        @lckr/jupyterlab_variableinspector v0.5.1  enabled  OK
        nbdime-jupyterlab v2.0.1  enabled  OK
```

Here is the output for jupyter troubleshoot:

```
root@ead842543f65:/home/test# jupyter troubleshoot
WARNING: You are using pip version 20.2.2; however, version 20.2.4 is available.
You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
$PATH:
        /usr/local/sbin
        /usr/local/bin
        /usr/sbin
        /usr/bin
        /sbin
        /bin

sys.path:
        /usr/local/bin
        /usr/lib/python38.zip
        /usr/lib/python3.8
        /usr/lib/python3.8/lib-dynload
        /usr/local/lib/python3.8/dist-packages
        /usr/lib/python3/dist-packages

sys.executable:
        /usr/bin/python3

sys.version:
        3.8.5 (default, Jul 28 2020, 12:59:40)
        [GCC 9.3.0]

platform.platform():
        Linux-4.19.76-linuxkit-x86_64-with-glibc2.29

which -a jupyter:
        /usr/local/bin/jupyter

pip list:
        Package                           Version
        --------------------------------- ---------
        alembic                           1.4.3
        argon2-cffi                       20.1.0
        astroid                           2.4.2
        async-generator                   1.10
        attrs                             20.2.0
        autopep8                          1.5.4
        backcall                          0.2.0
        bcrypt                            3.2.0
        bleach                            3.2.1
        certifi                           2020.6.20
        certipy                           0.1.3
        cffi                              1.14.3
        chardet                           3.0.4
        colorama                          0.4.4
        cryptography                      3.1.1
        dbus-python                       1.2.16
        decorator                         4.4.2
        defusedxml                        0.6.0
        entrypoints                       0.3
        flake8                            3.8.4
        gitdb                             4.0.5
        GitPython                         3.1.9
        idna                              2.10
        ipykernel                         5.3.4
        ipython                           7.18.1
        ipython-genutils                  0.2.0
        isort                             5.6.4
        jedi                              0.17.2
        Jinja2                            2.11.2
        json5                             0.9.5
        jsonschema                        3.2.0
        jupyter-client                    6.1.7
        jupyter-contrib-core              0.3.3
        jupyter-contrib-nbextensions      0.5.1
        jupyter-core                      4.6.3
        jupyter-highlight-selected-word   0.2.0
        jupyter-latex-envs                1.4.6
        jupyter-lsp                       0.9.2
        jupyter-nbextensions-configurator 0.4.1
        jupyter-telemetry                 0.1.0
        jupyterhub                        1.2.0b1
        jupyterhub-nativeauthenticator    0.0.5
        jupyterlab                        2.2.8
        jupyterlab-git                    0.22.1
        jupyterlab-latex                  2.0.0
        jupyterlab-pygments               0.1.2
        jupyterlab-server                 1.2.0
        lazy-object-proxy                 1.4.3
        lxml                              4.6.1
        Mako                              1.1.3
        MarkupSafe                        1.1.1
        mccabe                            0.6.1
        mistune                           0.8.4
        nbclient                          0.5.1
        nbconvert                         6.0.7
        nbdime                            2.1.0
        nbformat                          5.0.8
        nest-asyncio                      1.4.1
        notebook                          6.1.4
        numpy                             1.17.4
        oauthlib                          3.1.0
        onetimepass                       1.0.1
        packaging                         20.4
        pamela                            1.0.0
        pandocfilters                     1.4.2
        parso                             0.7.1
        pexpect                           4.8.0
        pickleshare                       0.7.5
        pip                               20.2.2
        pluggy                            0.13.1
        prometheus-client                 0.8.0
        prompt-toolkit                    3.0.8
        ptyprocess                        0.6.0
        pycodestyle                       2.6.0
        pycparser                         2.20
        pycurl                            7.43.0.2
        pydocstyle                        5.1.1
        pyflakes                          2.2.0
        Pygments                          2.7.1
        PyGObject                         3.36.0
        pylint                            2.6.0
        pyOpenSSL                         19.1.0
        pyparsing                         2.4.7
        PyQt5                             5.14.1
        pyrsistent                        0.17.3
        python-dateutil                   2.8.1
        python-editor                     1.0.4
        python-json-logger                2.0.1
        python-jsonrpc-server             0.4.0
        python-language-server            0.35.1
        PyYAML                            5.3.1
        pyzmq                             19.0.2
        requests                          2.24.0
        rope                              0.18.0
        ruamel.yaml                       0.16.12
        ruamel.yaml.clib                  0.2.2
        Send2Trash                        1.5.0
        setuptools                        50.3.1
        sip                               4.19.21
        six                               1.15.0
        smmap                             3.0.4
        snowballstemmer                   2.0.0
        SQLAlchemy                        1.3.20
        swiglpk                           4.65.0
        terminado                         0.9.1
        testpath                          0.4.4
        toml                              0.10.1
        tornado                           6.0.4
        traitlets                         5.0.5
        ujson                             4.0.1
        urllib3                           1.25.10
        veusz                             3.0.1
        wcwidth                           0.2.5
        webencodings                      0.5.1
        wheel                             0.34.2
        wrapt                             1.12.1
        yapf                              0.30.0
```

**Comment by krassowski** [comment]
Context menu, see (slightly outdated, yet relevant) gif below:

![diagnostics_panel](https://user-images.githubusercontent.com/5832902/71786083-a040f900-2fff-11ea-9fdb-fe53bd99ac3d.gif)

**Comment by krassowski** [comment]
Thank you for the report and including the screenshot. I understand that you expected that there would be a command in the command palette. It is a good idea for a new feature.

**Comment by krassowski** [comment]
Actually, it should have shown up in there..

![Screenshot from 2020-10-25 10-23-41](https://user-images.githubusercontent.com/5832902/97104503-469fdd00-16ac-11eb-8eba-e8ba9fedf435.png)

You can access the browser console for logs of interface errors and warnings, see [How_to_open_the_devtools_in_your_browser](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_are_browser_developer_tools#How_to_open_the_devtools_in_your_browser)

**Comment by stefaneidelloth** [comment]
Does not show up in context menu of cell either:
![image](https://user-images.githubusercontent.com/12675339/97104727-63d9a900-16b6-11eb-80c8-7a4a71f48093.png)

**Comment by krassowski** [comment]
Thank you. This is intriguing as there is nothing unexpected in the logs. I am sorry it does not work as expected. Would you mind taking a screenshot of the statusbar item which should look somewhat like that:

![Screenshot from 2020-10-25 10-47-23](https://user-images.githubusercontent.com/5832902/97104954-89af7f80-16af-11eb-9d06-8f33cdd7f0c9.png)

**Comment by stefaneidelloth** [comment]
![image](https://user-images.githubusercontent.com/12675339/97105098-e19eb400-16b8-11eb-969a-36fd40576544.png)

I restarted the kernel and got some more output. 


notebook.js:25 LSP: kernel was shut down
notebook.js:30 LSP: Changed to python kernel, reconnecting
adapter.js:307 LSP: will connect using language: python
connection_manager.js:159 LSP: connection requested {virtual_document: F, language: "python", document_path: "projects/hello_world.ipynb"}
connection_manager.js:94 LSP: Connection Socket {virtual_document: F, language: "python", document_path: "projects/hello_world.ipynb"}
14react_devtools_backend.js:2450 TypeError: Cannot read property 'editor' of undefined
    at F.get_editor_at_source_line (document.js:490)
    at Proxy.get_editor_at_root_line (codemirror_editor.js:272)
    at Proxy.getTokenAt (codemirror_editor.js:279)
    at pt._updateUnderlineAndTooltip (hover.js:257)
    at pt.updateUnderlineAndTooltip (hover.js:109)
    at HTMLDivElement.<anonymous> (hover.js:153)
overrideMethod @ react_devtools_backend.js:2450
Promise.catch (async)
(anonymous) @ hover.js:159
5react_devtools_backend.js:2450 is_visible failed TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.context_from_active_document (notebook.js:139)
    at w.get_context (command_manager.js:116)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
is_visible @ command_manager.js:129
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
u @ index.es6.js:5848
t.open @ index.es6.js:5108
e.open @ index.es6.js:6271
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
5react_devtools_backend.js:2450 is_visible failed TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.context_from_active_document (notebook.js:139)
    at w.get_context (command_manager.js:116)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at n.e.createItemClass (index.es6.js:5630)
overrideMethod @ react_devtools_backend.js:2450
is_visible @ command_manager.js:129
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
e.createItemClass @ index.es6.js:5630
e.renderItem @ index.es6.js:5536
t.onUpdateRequest @ index.es6.js:5193
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
u @ index.es6.js:5848
t.open @ index.es6.js:5108
e.open @ index.es6.js:6271
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 contextMenu is attached, but could not get the context TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.get_context_from_context_menu (adapter.js:399)
    at w.get_context (command_manager.js:108)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
get_context @ command_manager.js:111
is_visible @ command_manager.js:123
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 is_visible failed TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.context_from_active_document (notebook.js:139)
    at w.get_context (command_manager.js:116)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
is_visible @ command_manager.js:129
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 contextMenu is attached, but could not get the context TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.get_context_from_context_menu (adapter.js:399)
    at w.get_context (command_manager.js:108)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
get_context @ command_manager.js:111
is_visible @ command_manager.js:123
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 is_visible failed TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.context_from_active_document (notebook.js:139)
    at w.get_context (command_manager.js:116)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
is_visible @ command_manager.js:129
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 contextMenu is attached, but could not get the context TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.get_context_from_context_menu (adapter.js:399)
    at w.get_context (command_manager.js:108)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
get_context @ command_manager.js:111
is_visible @ command_manager.js:123
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 is_visible failed TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.context_from_active_document (notebook.js:139)
    at w.get_context (command_manager.js:116)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
is_visible @ command_manager.js:129
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 contextMenu is attached, but could not get the context TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.get_context_from_context_menu (adapter.js:399)
    at w.get_context (command_manager.js:108)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
get_context @ command_manager.js:111
is_visible @ command_manager.js:123
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 is_visible failed TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.context_from_active_document (notebook.js:139)
    at w.get_context (command_manager.js:116)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
is_visible @ command_manager.js:129
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 contextMenu is attached, but could not get the context TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.get_context_from_context_menu (adapter.js:399)
    at w.get_context (command_manager.js:108)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
get_context @ command_manager.js:111
is_visible @ command_manager.js:123
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 is_visible failed TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.context_from_active_document (notebook.js:139)
    at w.get_context (command_manager.js:116)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:6173)
    at Object.c [as computeCollapsed] (index.es6.js:5802)
overrideMethod @ react_devtools_backend.js:2450
is_visible @ command_manager.js:129
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:6173
c @ index.es6.js:5802
t.onUpdateRequest @ index.es6.js:5187
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
react_devtools_backend.js:2450 TypeError: Cannot read property 'editor' of undefined
    at F.get_editor_at_source_line (document.js:490)
    at Proxy.get_editor_at_root_line (codemirror_editor.js:272)
    at Proxy.getTokenAt (codemirror_editor.js:279)
    at Proxy.get_token_at (codemirror_editor.js:122)
    at _ (command_manager.js:6)
    at w.is_enabled (command_manager.js:97)
    at isEnabled (command_manager.js:28)
    at e.isEnabled (index.es6.js:302)
    at e.get [as isEnabled] (index.es6.js:6144)
    at n.e.createItemClass (index.es6.js:5618)
overrideMethod @ react_devtools_backend.js:2450
g @ index.es6.js:389
x @ index.es6.js:445
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.clearItems @ index.es6.js:5080
e.open @ index.es6.js:6257
evtContextMenu @ frontend.js:98
e.handleEvent @ index.es6.js:314
16react_devtools_backend.js:2450 TypeError: Cannot read property 'editor' of undefined
    at F.get_editor_at_source_line (document.js:490)
    at Proxy.get_editor_at_root_line (codemirror_editor.js:272)
    at Proxy.getTokenAt (codemirror_editor.js:279)
    at pt._updateUnderlineAndTooltip (hover.js:257)
    at pt.updateUnderlineAndTooltip (hover.js:109)
    at HTMLDivElement.<anonymous> (hover.js:153)
overrideMethod @ react_devtools_backend.js:2450
Promise.catch (async)
(anonymous) @ hover.js:159
150react_devtools_backend.js:2450 is_visible failed TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at bn.get_context (adapter.js:385)
    at bn.context_from_active_document (notebook.js:139)
    at w.get_context (command_manager.js:116)
    at w.is_visible (command_manager.js:123)
    at isVisible (command_manager.js:29)
    at e.isVisible (index.es6.js:332)
    at e.get [as isVisible] (index.es6.js:4689)
    at c (index.es6.js:4388)
overrideMethod @ react_devtools_backend.js:2450
is_visible @ command_manager.js:129
isVisible @ command_manager.js:29
e.isVisible @ index.es6.js:332
get @ index.es6.js:4689
c @ index.es6.js:4388
i @ index.es6.js:4350
t.onUpdateRequest @ index.es6.js:3859
e.processMessage @ index.es6.js:1341
x @ index.es6.js:442
n @ index.es6.js:178
k @ index.es6.js:492
requestAnimationFrame (async)
C @ index.es6.js:461
a @ index.es6.js:229
e.update @ index.es6.js:1197
t.refresh @ index.es6.js:3789
t.handleEvent @ index.es6.js:3810
react_devtools_backend.js:2450 LSP: Connect timed out for python
overrideMethod @ react_devtools_backend.js:2450
connect @ connection_manager.js:167
async function (async)
connect @ connection_manager.js:160
connect @ adapter.js:313
connect_document @ adapter.js:201
reload_connection @ adapter.js:148
on_kernel_changed @ notebook.js:32
async function (async)
on_kernel_changed @ notebook.js:29
m @ index.es6.js:459
c @ index.es6.js:416
e.emit @ index.es6.js:99
_handleNewSession @ sessioncontext.js:550
_changeKernel @ sessioncontext.js:500
async function (async)
_changeKernel @ sessioncontext.js:481
changeKernel @ sessioncontext.js:316
async function (async)
changeKernel @ sessioncontext.js:315
restart @ sessioncontext.js:733
async function (async)
restart @ sessioncontext.js:727
execute @ index.js:888
e.execute @ index.es6.js:357
t.triggerActiveItem @ index.es6.js:4982
t._evtMouseUp @ index.es6.js:5318
t.handleEvent @ index.es6.js:5128
react_devtools_backend.js:2450 TypeError: Cannot read property 'on' of undefined
    at Wt.register (codemirror.js:47)
    at Wt.register (diagnostics.js:263)
    at new dn (editor_adapter.js:12)
    at bn.create_adapter (adapter.js:351)
    at bn.connect_adapter (adapter.js:290)
    at bn.on_connected (adapter.js:181)
    at bn.connect (adapter.js:314)
    at async bn.connect_document (adapter.js:201)
overrideMethod @ react_devtools_backend.js:2450
Promise.catch (async)
connect_document @ adapter.js:201
reload_connection @ adapter.js:148
on_kernel_changed @ notebook.js:32
async function (async)
on_kernel_changed @ notebook.js:29
m @ index.es6.js:459
c @ index.es6.js:416
e.emit @ index.es6.js:99
_handleNewSession @ sessioncontext.js:550
_changeKernel @ sessioncontext.js:500
async function (async)
_changeKernel @ sessioncontext.js:481
changeKernel @ sessioncontext.js:316
async function (async)
changeKernel @ sessioncontext.js:315
restart @ sessioncontext.js:733
async function (async)
restart @ sessioncontext.js:727
execute @ index.js:888
e.execute @ index.es6.js:357
t.triggerActiveItem @ index.es6.js:4982
t._evtMouseUp @ index.es6.js:5318
t.handleEvent @ index.es6.js:5128
react_devtools_backend.js:2450 Connection for projects/hello_world.ipynb was not opened
overrideMethod @ react_devtools_backend.js:2450
connect_document @ adapter.js:209
async function (async)
connect_document @ adapter.js:201
reload_connection @ adapter.js:148
on_kernel_changed @ notebook.js:32
async function (async)
on_kernel_changed @ notebook.js:29
m @ index.es6.js:459
c @ index.es6.js:416
e.emit @ index.es6.js:99
_handleNewSession @ sessioncontext.js:550
_changeKernel @ sessioncontext.js:500
async function (async)
_changeKernel @ sessioncontext.js:481
changeKernel @ sessioncontext.js:316
async function (async)
changeKernel @ sessioncontext.js:315
restart @ sessioncontext.js:733
async function (async)
restart @ sessioncontext.js:727
execute @ index.js:888
e.execute @ index.es6.js:357
t.triggerActiveItem @ index.es6.js:4982
t._evtMouseUp @ index.es6.js:5318
t.handleEvent @ index.es6.js:5128
100react_devtools_backend.js:2450 TypeError: Cannot read property 'editor' of undefined
    at F.get_editor_at_source_line (document.js:490)
    at Proxy.get_editor_at_root_line (codemirror_editor.js:272)
    at Proxy.getTokenAt (codemirror_editor.js:279)
    at pt._updateUnderlineAndTooltip (hover.js:257)
    at pt.updateUnderlineAndTooltip (hover.js:109)
    at HTMLDivElement.<anonymous> (hover.js:153)
overrideMethod @ react_devtools_backend.js:2450
Promise.catch (async)
(anonymous) @ hover.js:159
react_devtools_backend.js:2450 Could not get highlights: TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at onCursorActivity (highlights.js:83)
overrideMethod @ react_devtools_backend.js:2450
onCursorActivity @ highlights.js:90
async function (async)
onCursorActivity @ highlights.js:61
r @ codemirror_editor.js:159
En @ codemirror.js:2038
Mn @ codemirror.js:2047
Ci @ codemirror.js:3787
(anonymous) @ codemirror.js:3924
26react_devtools_backend.js:2450 TypeError: Cannot read property 'editor' of undefined
    at F.get_editor_at_source_line (document.js:490)
    at Proxy.get_editor_at_root_line (codemirror_editor.js:272)
    at Proxy.getTokenAt (codemirror_editor.js:279)
    at pt._updateUnderlineAndTooltip (hover.js:257)
    at pt.updateUnderlineAndTooltip (hover.js:109)
    at HTMLDivElement.<anonymous> (hover.js:153)
overrideMethod @ react_devtools_backend.js:2450
Promise.catch (async)
(anonymous) @ hover.js:159
react_devtools_backend.js:2450 Could not get highlights: TypeError: Cannot read property 'virtual_line' of undefined
    at F.virtual_position_at_document (document.js:224)
    at Proxy.root_position_to_virtual_position (codemirror_editor.js:142)
    at onCursorActivity (highlights.js:83)
overrideMethod @ react_devtools_backend.js:2450
onCursorActivity @ highlights.js:90
async function (async)
onCursorActivity @ highlights.js:61
r @ codemirror_editor.js:159
En @ codemirror.js:2038
Mn @ codemirror.js:2047
Ci @ codemirror.js:3787
(anonymous) @ codemirror.js:3924
94react_devtools_backend.js:2450 TypeError: Cannot read property 'editor' of undefined
    at F.get_editor_at_source_line (document.js:490)
    at Proxy.get_editor_at_root_line (codemirror_editor.js:272)
    at Proxy.getTokenAt (codemirror_editor.js:279)
    at pt._updateUnderlineAndTooltip (hover.js:257)
    at pt.updateUnderlineAndTooltip (hover.js:109)
    at HTMLDivElement.<anonymous> (hover.js:153)
overrideMethod @ react_devtools_backend.js:2450
Promise.catch (async)
(anonymous) @ hover.js:159

**Comment by stefaneidelloth** [comment]
Hm. Today the diagnostics panel works. Don't know why. I just cloned some git repository but don't guess that's the reason.
I'll let you know if I find out more.

**Comment by krassowski** [comment]
Is it working consistently well since?

**Comment by stefaneidelloth** [comment]
Thank you for asking back. Unfortunately not. I could share my screen a gotomeeting session if that would help to indentify the underlying issue.
![image](https://user-images.githubusercontent.com/12675339/97959480-be0df480-1daf-11eb-8781-761c771d3862.png)

**Comment by krassowski** [comment]
Thanks for the update! Btw, just released python-language-server 0.36 requires jedi 0.17.2.
